### ISSUE TITLE
Apple Health Permissions Bug(?)

### ISSUE THREAD

**Issue by marionbarker** [LoopKit/Loop#1703]
**Describe the bug**
One of my test phones on which I was running Loop-dev got into a state where Health permissions were not set properly and I could not set them. Symptom was no charts were showing on main Loop Screen and CGM data did not display even though pump (simulated) was operating nominally.

**To Reproduce**
1. I do not know how I got into this state but I could not get out of it - I may have deleted old health data as part of my testing
2. I resolved the problem by deleting the Loop-dev app from the phone
3. When I installed a fresh version of the app, the onboarding provided the screen needed to add health permissions

**Expected behavior**
I expect to be able to go the the Health app and set up permissions if they get turned off without having to completely delete the Loop app.

**Loop Report, Xcode Log and Screenshots**
Attached at end of message.

**Phone**
 - iPhone 8
 - OS Version: 15.4 and then updated to 15.5 (no change)

**Loop Version**
 - Loop-dev: Workspace git revisions: probably started with commit 6286f61, then updated to efd95a1 (no change)

**CGM**
 - Tried Simulator, Dexcom Share and Nightscout as a CGM (no change)

**Pump**
 - Using rPi DASH simulator (which was working but CGM data did not display)

**Additional context**
Error messages as seen in the Xcode debug log are of this type:
```
2022-05-29 08:20:36.067336-0700 Loop[84623:7062325] [HealthStoreUnitCache] Error fetching unit for HKQuantityTypeIdentifierBloodGlucose: Error Domain=com.apple.healthkit Code=5 "Authorization status is not determined for all types provided." UserInfo={NSLocalizedDescription=Authorization status is not determined for all types provided.}
```

Loop Report: 
[Loop Report 2022-05-29 090328-0700.md](https://github.com/LoopKit/Loop/files/8798263/Loop.Report.2022-05-29.090328-0700.md)

Xcode Debug Log:
[20220529_0820_xLog_HealthNotWorking.txt](https://github.com/LoopKit/Loop/files/8798266/20220529_0820_xLog_HealthNotWorking.txt)

Health App Screen:
![IMG_7B4E5DE88C09-1](https://user-images.githubusercontent.com/19607791/170991593-bb464abc-672a-47ec-801d-1ac5c08a6f06.jpeg)

Health Permissions Screen:
![IMG_1983E078E6C1-1](https://user-images.githubusercontent.com/19607791/170991673-1b5e2112-3c4e-4d6c-b6e3-4d8e7a5adba1.jpeg)

**Comment by skimaniac** [comment]
In data sources and acces, touch edit on the top right, after that you can authorize loop
![65FCA990-A49E-4255-84BB-4CD47AE8EF55](https://user-images.githubusercontent.com/44727542/171000599-f9009d3c-ea40-4be4-8030-a2f86a810e06.jpeg)

**Comment by marionbarker** [comment]
I did post in zulipchat first to ask for help before I deleted the app. 
I'll put this in LoopDocs for future reference. 
Thanks much.

I'll close this issue.

**Comment by marionbarker** [comment]
Reopening - a second test phone is exhibiting this behavior.
I just plugged it in (after not using it for a while) and built:
LoopWorkspace commit efd95a1
Loop commit 28b5588

Tapped on the Edit button in graphic shown above and nothing happens.
In this case, Loop is shown as an app but it shows as `inactive` as seen in graphic below.

![IMG_97F6CB4606F4-1](https://user-images.githubusercontent.com/19607791/171039917-32dbec13-bee3-41fe-92cd-c701798b4caa.jpeg)

Charts are blank (graphic below). 
![IMG_ECE107C244DF-1](https://user-images.githubusercontent.com/19607791/171039942-13d03359-b315-4987-b428-4f0f8e6bff9c.jpeg)

**Comment by ps2** [comment]
The error message indicates that we have not asked the user for all the pertinent permissions. This is a Loop bug, independent of whatever health permissions you may have granted/denied Loop in the Health App. Any steps to reproduce would be very helpful in getting a fix for this implemented.

**Comment by marionbarker** [comment]
I just built this same version of Loop-dev over Loop-master on my third test phone.
When I onboarded, the health permissions were all disabled and I had to enable them (can't remember if that is typical when building Loop-dev over master).
I will attempt to build an older version of Loop-dev on the test phone that is currently without permissions and see if that helps.

**Comment by marionbarker** [comment]
Tried a few older commits for LoopWorkspace with no change to permission behavior. I backed up as far as 84191b5, but I know I've built that before and did not have this problem. 

Could it be related to Xcode 13.4?

**Comment by ps2** [comment]
I'm guessing the state occurs when you have a phone that was was running some previous version of Loop (or FreeAPS) and install dev over it?

**Comment by marionbarker** [comment]
No. I have 3 test phones. Two are labeled Dev and I used them for Loop-dev testing. One is labeled Free. I used the Free phone to build Loop master over FreeAPS-dev while running a real DASH pod to make sure that worked. Building Loop-dev over that version of Loop master got the permissions right.

**Comment by marionbarker** [comment]
On the Loop-dev phone (iPhone 8) where I deleted the app and rebuilt to go through onboarding, I have deleted all health data and disabled permissions for Loop to read or write health data and the app appears to be happy and charts are showing.

For this phone, if I do health sources&data and tap on loop, I get this screen (which I expect). 

![IMG_079A3D13E3CE-1](https://user-images.githubusercontent.com/19607791/171055015-83c5323c-3f5d-4c80-a73b-efa0b50839d5.jpeg)

I do not get that with the other Loop-dev phone (6s).

**Comment by marionbarker** [comment]
Another report that looks very similar.  This one is with an active user and a real pod.
https://loop.zulipchat.com/#narrow/stream/144182-development/topic/Loop.20Display.20Gone/near/284412297

**Comment by holmfirth** [comment]
xcode text from my build which doesn't allow notifications to be enabled

[Build Loop (Workspace)_2022-06-01T14-15-42.txt.zip](https://github.com/LoopKit/Loop/files/8814968/Build.Loop.Workspace._2022-06-01T14-15-42.txt.zip)

**Comment by holmfirth** [comment]
and issue report from the same.

[Loop Report 2022-06-01 15:21:54+01:00.md](https://github.com/LoopKit/Loop/files/8815011/Loop.Report.2022-06-01.15.21.54%2B01.00.md)

**Comment by ps2** [comment]
Having steps to reproduce this would be very useful.

**Comment by holmfirth** [comment]
did a full build from scratch last week (never had Loop on phone), then did a pull yesterday to update to latest version. No notifications, so deleted app from phone, deleted build files from Mac and started from screatch. Built and installed OK, but no onboarding needed on any build. I wish I could tell you more but that's it! At no point have I deleted Loop data from Apple health - should I do that?

**Comment by holmfirth** [comment]
Macbook Air M1, Monterey 12.4, Xcover Version 13.4 (13F17a), iOS15.5.

**Comment by britnischipper** [comment]
https://mail.google.com/mail/u/0?ui=2&ik=548c8844c9&attid=0.1&permmsgid=msg-f:1734292055293556429&th=1811716f848df6cd&view=att&disp=safe

**Comment by britnischipper** [comment]
[Build Error 1.pdf](https://github.com/LoopKit/Loop/files/8841147/Build.Error.1.pdf)

**Comment by marionbarker** [comment]
My test phone running eros_new_ui is now exhibiting the Apple HealthKit Permission issue.
Loop Report attached below.
```quote
2022-06-10 12:40:35.909361-0700 Loop[21276:1013671] [HealthStoreUnitCache] Error fetching unit for HKQuantityTypeIdentifierBloodGlucose: Error Domain=com.apple.healthkit Code=5 "Authorization status is not determined for all types provided." UserInfo={NSLocalizedDescription=Authorization status is not determined for all types provided.}
```

Details:
After running tests with the eros_new_ui branch and reporting issues yesterday, I locked the phone and left it alone.
When I unlocked today, prior to building, it was stuck on Share Values from yesterday. Didn't think too much of it.
Did the fetch, pull recurse for eros_new_ui and built to the phone.
No display is happening on the main screen but other features appear to work.
The Eros pod is still connected (I can make it beep).
I can bolus.
Loop must be getting the values from CGM, because it recommended a bolus (CGM going up slightly after lunch) and the bolus went through.
I set a manual temp basal of 30 U/hr and I can hear the pod clicking.
Canceled the MTB.
[Loop Report 2022-06-10 12:47:39-07:00.md](https://github.com/LoopKit/Loop/files/8881907/Loop.Report.2022-06-10.12.47.39-07.00.md)
